source/class/qx/ui/tabview/TabView.js

```diff
@@ -637,6 +623,32 @@ qx.Class.define("qx.ui.tabview.TabView", {
       closeButton.reset();
 
       this.remove(page);
+    },
+
+    __onPageChangeTabVisibility() {
+      this.__updateFirstLastTabStates();
+    },
+
+    __updateFirstLastTabStates() {
+      let firstPage = null;
+      let lastPage = null;
+
+      for (let pages = this.getChildren(), i = 0; i < pages.length; i++) {
```

Comment:
Could you move `let pages = this.getChildren()` out of cycle on separate line, please? Several instructions on one line - looks like a bad smell.

Reply:
we use that pattern in a lot of places - I'm counting 290 `for` loops with multiple variables declared, one for the array and one for the index.  My regex is `for \((var|let) [^;]+,`

Reply:
Ok, I won't insist. Just noticed. My view on good code is obviously different. May fix it, it is up to you.